keen/api.py

```diff
@@ -126,12 +127,70 @@ def post_events(self, events):
         self._error_handling(response)
         return self._get_response_json(response)
 
+
+    def _order_by_is_valid_or_none(self, params):
+        """
+        Validates that a given order_by has proper syntax.
+
+        :param params: Query params.
+        :return: Returns True if either no order_by is present, or if the order_by is well-formed.
+        """
+        if not "order_by" in params or not params["order_by"]:
+            return True
+
+        def _order_by_dict_is_not_well_formed(d):
+            if not isinstance(d, dict):
+                # Bad type.
+                return True
+            if "property_name" in d and d["property_name"]:
+                if "direction" in d and not (d["direction"] == "ASC" or d["direction"] == "DESC"):
+                    # Bad direction provided.
+                    return True
+                for k in d:
+                    if k != "property_name" and k != "direction":
+                        # Unexpected key.
+                        return True
+                # Everything looks good!
+                return False
+            # Missing required key.
+            return True
+
+        # order_by is converted to a list before this point if it wasn't one before.
+        order_by_list = json.loads(params["order_by"])
+
+        for order_by in order_by_list:
```

Comment:
Should we explicitly make sure this list is of length 1? Or is there a legitimate reason it could have multiple entries? At this point it also shouldn't be empty, so just grabbing `order_by_list[0]` should be fine, right?

Reply:
Oh wait, now I see that you can have multiple orderings, so really `order_by` can be a list or a dict, but the comments don't seem to indicate that...so I'd say we have to change the comment everywhere it's pasted so that it makes that clear :/

Reply:
Yeah, I will update the comments.